docs/user-manual/scripting/script-lifecycle.md

```diff
@@ -0,0 +1,368 @@
+---
+title: Script Lifecycle
+sidebar_position: 2
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+Every script instance you attach to an Entity in PlayCanvas goes through a well-defined lifecycle. Understanding this lifecycle is crucial because it dictates when your code runs and how it can interact with the rest of your application. PlayCanvas provides specific functions, called lifecycle methods, that you can define in your script. The engine will automatically call these methods at the appropriate times.
+
+Think of it like the stages in an actor's performance: preparing backstage (`initialize`), performing on stage (`update`), and taking a final bow (`destroy` event).
+
+```mermaid
+graph TD
+    A["<code>initialize()</code>"] --> B["<code>postInitialize()</code>"]
+    B --> C["<code>update(dt)</code>"]
+    C --> D["<code>postUpdate(dt)</code>"]
+    D --> E["<code>destroy</code> handler"]
+    %% Loop back to update
+    D --> C
+
+    subgraph Script Lifecycle Sequence
+        A
+        B
+        C
+        D
+        E
+    end
+```
+
+## Lifecycle Methods
+
+Let's break down each of the key lifecycle methods.
+
+### `initialize()`
+
+**When it's called:**
+
+* Once per script instance.
+* After the script instance is created and its Entity is enabled.
+* After all its Script Attributes have been parsed and assigned their initial values (either defaults or values set in the Editor).
+
+**Purpose:**
+
+* This is your script's primary setup or "constructor-like" phase.
```

Comment:
Probably need to add a section/info on ESM Script constructors (can you use them, when you'd use them vs the initialize function).